### ISSUE TITLE
Shader gamma correction does not match Lua side by default

### ISSUE THREAD

**Issue by Krunklehorn** [love2d/love#1668]
TL;DR at the bottom...

I stumbled upon this issue while setting up a project to learn about banding and dithering.

The top rows for each gradient are indexed from an array of color values that have been gamma corrected from the Lua side using love.math.gammaToLinear(). The bottom rows come from color values created in shader and outputted through the gammaCorrectColor() function. Notice the imprecision in the darker areas on the far left...

![love_2F6uT87h53](https://user-images.githubusercontent.com/42424291/105804393-b48ac780-5f6d-11eb-9b53-f1fa600abca8.png)

This issue comes from wrap_GraphicsShader.lua where 'gammaCorrectColor' and 'unGammaCorrectColor' are aliased to the 'fast' variants defined just above it.

https://github.com/love2d/love/blob/48a13d443b739d98f6ba95b6db73f102b5b787ec/src/modules/graphics/wrap_GraphicsShader.lua#L177-L182

These fast variants do not match the functions defined by MathModule.cpp on the Lua side, causing a mismatch when attempting manual gamma correction in shader.

https://github.com/love2d/love/blob/d2c42bd74694fe95c46a11b60dfc7b037e381d99/src/modules/math/MathModule.cpp#L183-L189

For my test case I was rendering straight to the screen, but the issue is not resolved if I render to a Canvas with the "rgba32f" format.

I also tried manually calling love.math.gammaToLinear() while swapping shader:sendColor() for shader:send() just to see if it made a difference. The indexed colors still appear fine.

Once I dug through the source code and found this alias on the shader side, I forced the 'Precise' variant by calling gammaCorrectColorPrecise() and presto, the rows matched.

TL;DR...
By default, the precise variant of gamma correction should be used on the shader side, not the fast one. Otherwise, the documentation should be updated to notify users of this difference.

**Comment by slime73** [comment]
I updated the https://love2d.org/wiki/love.graphics.isGammaCorrect wiki page with a bit more information.
